src/hb-ot-layout.cc

```diff
@@ -1009,43 +1027,109 @@ hb_ot_layout_get_size_params (hb_face_t    *face,
 			      unsigned int *range_start,       /* OUT.  May be nullptr */
 			      unsigned int *range_end          /* OUT.  May be nullptr */)
 {
-  const OT::GPOS &gpos = _get_gpos (face);
-  const hb_tag_t tag = HB_TAG ('s','i','z','e');
-
-  unsigned int num_features = gpos.get_feature_count ();
-  for (unsigned int i = 0; i < num_features; i++)
+  hb_tag_t tag = HB_TAG ('s','i','z','e');
+  const OT::FeatureParams &feature_params =
+    _get_gsubgpos_matched_feature_params (face, tag);
+  if (&feature_params != &OT::Null (OT::FeatureParams))
   {
-    if (tag == gpos.get_feature_tag (i))
+    const OT::FeatureParamsSize params = feature_params.get_size_params (tag);
+    if (params.designSize)
     {
-      const OT::Feature &f = gpos.get_feature (i);
-      const OT::FeatureParamsSize &params = f.get_feature_params ().get_size_params (tag);
-
-      if (params.designSize)
-      {
 #define PARAM(a, A) if (a) *a = params.A
-	PARAM (design_size, designSize);
-	PARAM (subfamily_id, subfamilyID);
-	PARAM (subfamily_name_id, subfamilyNameID);
-	PARAM (range_start, rangeStart);
-	PARAM (range_end, rangeEnd);
+      PARAM (design_size, designSize);
+      PARAM (subfamily_id, subfamilyID);
+      PARAM (subfamily_name_id, subfamilyNameID);
+      PARAM (range_start, rangeStart);
+      PARAM (range_end, rangeEnd);
 #undef PARAM
 
-	return true;
-      }
+      return true;
     }
   }
 
-#define PARAM(a, A) if (a) *a = 0
-  PARAM (design_size, designSize);
-  PARAM (subfamily_id, subfamilyID);
-  PARAM (subfamily_name_id, subfamilyNameID);
-  PARAM (range_start, rangeStart);
-  PARAM (range_end, rangeEnd);
-#undef PARAM
+#define UNPARAM(a) if (a) *a = 0
+  UNPARAM (design_size);
+  UNPARAM (subfamily_id);
+  UNPARAM (subfamily_name_id);
+  UNPARAM (range_start);
+  UNPARAM (range_end);
+#undef UNPARAM
 
   return false;
 }
 
+/**
+ * hb_ot_layout_get_feature_parameters:
+ *
+ * Since: REPLACEME
+ **/
+hb_bool_t
+hb_ot_layout_get_feature_parameters (hb_face_t       *face,
+				     hb_tag_t        tag,
+				     unsigned int    *label_id             /* OUT.  May be nullptr */,
+				     unsigned int    *tooltip_id           /* OUT.  May be nullptr */,
+				     unsigned int    *sample_id            /* OUT.  May be nullptr */,
+				     unsigned int    *num_named_parameters /* OUT.  May be nullptr */,
+				     unsigned int    *first_param_id       /* OUT.  May be nullptr */,
+				     unsigned int    *char_count           /* OUT.  May be nullptr */,
+				     hb_codepoint_t **characters           /* OUT.  May be nullptr */)
+{
+  const OT::FeatureParams &feature_params =
+    _get_gsubgpos_matched_feature_params (face, tag);
+  if (&feature_params != &OT::Null (OT::FeatureParams))
+  {
+    if ((tag & 0xFFFF0000u) == HB_TAG ('s','s','\0','\0')) /* ssXX */
+    {
```

Comment:
Instead of repeating the tag comparison that `get_stylistic_set_params()` already does, what about calling it unconditionally and checking if the result is not `Null`?.

Reply:
Done